### PR TITLE
Move API from core repo with history

### PR DESCRIPTION
**Pull request by dyladan**
Please pay special attention to this commit: [`cafb14f` (#1)](https://github.com/open-telemetry/opentelemetry-js-api/pull/1/commits/cafb14f6598e97063a154761b4c55f0ea19479ea)

Everything else is from the old repo

### CODE REVIEW

**Comment by dyladan** [comment]
@bogdandrutu I'm trying to move history from another repository and it looks like some of the contributions happened using the old CLA bot and they don't translate to the new one. Is there anything we can do here?

**Review by vmarchaud — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -10,17 +10,18 @@
...
-    "compile": "tsc --build",
+    "build": "npm run compile",
     "clean": "tsc --build --clean",
-    "test": "nyc ts-mocha -p tsconfig.json test/**/*.test.ts",
-    "test:browser": "nyc karma start --single-run",
-    "codecov": "nyc report --reporter=json && codecov -f coverage/*.json -p ../../",
     "codecov:browser": "nyc report --reporter=json && codecov -f coverage/*.json -p ../../",
-    "build": "npm run compile",
-    "lint": "eslint src test --ext .ts",
+    "codecov": "nyc report --reporter=json && codecov -f coverage/*.json -p ../../",
```
Comment (vmarchaud) [inline package.json]: i think we can remove the `-p ../..`, it was usefull to move the root of the repo but by default we already are in this repo

File: package.json [hunk 2]
```diff
@@ -42,23 +41,26 @@
...
     "@types/mocha": "8.2.0",
     "@types/node": "14.14.20",
     "@types/sinon": "9.0.10",
     "@types/webpack-env": "1.16.0",
+    "@typescript-eslint/eslint-plugin": "^4.15.1",
+    "@typescript-eslint/parser": "^4.15.1",
     "codecov": "3.8.1",
-    "gts": "3.1.0",
+    "eslint": "^7.20.0",
+    "eslint-config-airbnb-base": "^14.2.1",
```
Comment (vmarchaud) [inline package.json:60]: is this actually used, i don't see in the eslint config ?

File: package.json (hunk 1 above)
Comment (vmarchaud) [reply package.json]: Sorry this is fixed later

Comment (dyladan) [reply package.json]: it's changed in a later commit

File: package.json (hunk 2 above)
Comment (dyladan) [reply package.json]: Fixed

**Review by obecny — APPROVED** [review]
lgtm, I guess we will have to make a few manual testing too after this is merged

**Comment by dyladan** [comment]
> lgtm, I guess we will have to make a few manual testing too after this is merged

What do you mean?

**Comment by obecny** [comment]
> > lgtm, I guess we will have to make a few manual testing too after this is merged
> 
> What do you mean?

checking out, build, copy built files over and simply check with core repo if all is working fine

**Comment by dyladan** [comment]
> > > lgtm, I guess we will have to make a few manual testing too after this is merged
> > 
> > 
> > What do you mean?
> 
> checking out, build, copy built files over and simply check with core repo if all is working fine

I've done this locally by using `npm pack` and installing it by tarball in the other repo. Things seemed to be OK but yes I think it is a good idea for others to check also.

**Pull request merged**